### PR TITLE
[zeek] - update package-spec to 2.10.0

### PR DESCRIPTION
**Pull request by taylor-swanson**
## What does this PR do?

- Update package-spec to 2.10.0
- Ensure many fields are arrays (tls fields, host.ip, etc)
- Generate sample events for many data streams

```
[git-generate]
go run github.com/andrewkroh/go-examples/ecs-update@latest -ecs-version=8.9.0 -ecs-git-ref=v8.9.0 -format-version=2.10.0 packages/zeek
```

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates elastic/security-team#5870

### CODE REVIEW

File: packages/zeek/data_stream/known_certs/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -99,13 +103,21 @@ processors:
         - 'CN=%{CN:tls.server.x509.subject.common_name}'
       pattern_definitions:
         CN: '[^,]+'
+  - set:
+      field: tls.server.x509.subject.common_name
+      value: ['{{{tls.server.x509.subject.common_name}}}']
+      if: ctx.tls?.server?.x509?.subject?.common_name instanceof String
```
Comment (efd6) [inline packages/zeek/data_stream/known_certs/elasticsearch/ingest_pipeline/default.yml:109]: Is there a reason for the difference in approaches between here and the kerberos datastream?

File: packages/zeek/data_stream/smb_files/sample_event.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            "info"
+        ]
+    },
+    "file": {
+        "accessed": "2017-10-09T16:13:19.607Z",
+        "created": "2017-10-09T16:13:19.607Z",
+        "ctime": "2017-10-09T16:13:19.607Z",
+        "mtime": "2017-10-09T16:13:19.607Z",
+        "name": "PSEXESVC.exe",
+        "path": "\\\\\\\\admin-pc\\\\ADMIN$\\PSEXESVC.exe",
```
Comment (efd6) [inline packages/zeek/data_stream/smb_files/sample_event.json:53]: This doesn't look right; the input is `"\u005c\u005cadmin-pc\u005cADMIN$"`/`"PSEXESVC.exe"` which should expand to `"\\\\admin-pc\\ADMIN$\\PSECESVC.exe"`. That the last, synthesised, bs is correct suggests that there is improper escaping going on here.

Yup. The processor is not using literal template staches:
```
  - set:
      field: file.path
      value: "{{zeek.smb_files.path}}\\{{zeek.smb_files.name}}"
      if: "ctx?.zeek?.smb_files?.path != null && ctx?.zeek?.smb_files?.name != null"
```
(should be "{{{zeek.smb_files.path}}}\\{{{zeek.smb_files.name}}}")

I imagine this is happening elsewhere, and in general there should be a change form `{{…}}` to `{{{…}}}`.

File: packages/zeek/data_stream/known_certs/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (taylor-swanson) [reply packages/zeek/data_stream/known_certs/elasticsearch/ingest_pipeline/default.yml:109]: That was an oops on my part, realigned everything to use the set processor in `keberos`, `ssl`, and `x509`.

File: packages/zeek/data_stream/smb_files/sample_event.json (hunk 2 above)
Comment (efd6) [reply packages/zeek/data_stream/smb_files/sample_event.json:53]: Yeah, that's fine with me.

File: packages/zeek/data_stream/known_certs/elasticsearch/ingest_pipeline/default.yml (hunk 1 above)
Comment (efd6) [reply packages/zeek/data_stream/known_certs/elasticsearch/ingest_pipeline/default.yml:109]: Cool. I like the newer idiom.

**Review by efd6 — APPROVED** [review]

**Pull request merged**